File: Quorum3/Library/Standard/Plugins/javascript/Libraries/Game/Graphics/ColorAttribute.js
function plugins_quorum_Libraries_Game_Graphics_ColorAttribute_(quorumObject) 
{
    var attributePlugin = quorumObject.Libraries_Game_Graphics_Attribute__.plugin_;
    
    this.DIFFUSE = attributePlugin.Register("diffuseColor");
    this.SPECULAR = attributePlugin.Register("specularColor");
    this.AMBIENT = attributePlugin.Register("ambientColor");
    this.EMISSIVE = attributePlugin.Register("emissiveColor");
    this.REFLECTION = attributePlugin.Register("reflectionColor");
    this.AMBIENT_LIGHT = attributePlugin.Register("ambientLightColor");
    this.FOG = attributePlugin.Register("fogColor");
    
    this.MASK = this.DIFFUSE | this.SPECULAR | this.AMBIENT | this.EMISSIVE 
        | this.REFLECTION | this.AMBIENT_LIGHT | this.FOG;
    
    this.SupportsAttribute$quorum_integer = function(type) 
    {
        return (type & this.MASK) !== 0;
    };
    
    this.GetDiffuseValue = function() 
    {
        return this.DIFFUSE;
    };
    
    this.GetSpecularValue = function() 
    {
        return this.SPECULAR;
    };
    
    this.GetAmbientValue = function() 
    {
        return this.AMBIENT;
    };
    
    this.GetEmissiveValue = function() 
    {
        return this.EMISSIVE;
    };
    
    this.GetReflectionValue = function() 
    {
        return this.REFLECTION;
    };
    
    this.GetAmbientLightValue = function() 
    {
        return this.AMBIENT_LIGHT;
    };
    
    this.GetFogValue = function() 
    {
        return this.FOG;
    };
    
    this.GetColorAttributeMask = function() 
    {
        return this.MASK;
    };
    
    this.GetAttributeType$quorum_text = function(alias)
    {
        attributePlugin.GetAttributeType$quorum_text(alias);
    };
    
    this.GetAttributeAlias$quorum_integer = function(type)
    {
        attributePlugin.GetAttributeAlias$quorum_integer(type);
    };
    
    this.Register = function(alias)
    {
        attributePlugin.Register(alias);
    };
}

File: Plugins/GamePlugin/JavaGame/src/plugins/quorum/Libraries/Game/Graphics/ColorAttribute.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Game.Graphics;



/**
 *
 * @author alleew
 */
public class ColorAttribute extends Attribute
{
    public java.lang.Object me_ = null;
    
    public final static int DIFFUSE = Attribute.RegisterStatic("diffuseColor");
    public final static int SPECULAR = Attribute.RegisterStatic("specularColor");
    public final static int AMBIENT = Attribute.RegisterStatic("ambientColor");
    public final static int EMISSIVE = Attribute.RegisterStatic("emissiveColor");
    public final static int REFLECTION = Attribute.RegisterStatic("reflectionColor");
    public final static int AMBIENT_LIGHT = Attribute.RegisterStatic("ambientLightColor");
    public final static int FOG = Attribute.RegisterStatic("fogColor");
    
    public final static int MASK = DIFFUSE | SPECULAR | AMBIENT | EMISSIVE | REFLECTION | AMBIENT_LIGHT | FOG;
    
    public boolean SupportsAttribute(int type)
    {
        return (type & MASK) != 0;
    }
    
    public int GetDiffuseValue()
    {
        return DIFFUSE;
    }
    
    public int GetSpecularValue()
    {
        return SPECULAR;
    }
    
    public int GetAmbientValue()
    {
        return AMBIENT;
    }
    
    public int GetEmissiveValue()
    {
        return EMISSIVE;
    }
    
    public int GetReflectionValue()
    {
        return REFLECTION;
    }
    
    public int GetAmbientLightValue()
    {
        return AMBIENT_LIGHT;
    }
    
    public int GetFogValue()
    {
        return FOG;
    }
    
    public int GetColorAttributeMask()
    {
        return MASK;
    }
}
